Menus/SettingsVideoGUI.cpp

```diff
@@ -0,0 +1,365 @@
+		m_PresetResolutionMessageLabel = dynamic_cast<GUILabel *>(m_GUIControlManager->GetControl("LabelPresetResolutonValidation"));
+
+		if (g_FrameMan.GetMaxResX() == 1366 && g_FrameMan.GetMaxResY() == 768) {
+			m_PresetResolutionMessageLabel->SetText("1366x768 is not supported by the borderless driver for reasons unknown to man.\nPlease use the custom resolution controls with the dedicated driver to set it.");
+		} else {
+			m_PresetResolutionMessageLabel->SetVisible(false);
+		}
+		PopulateResolutionsComboBox();
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void SettingsVideoGUI::CreateCustomResolutionBox() {
+		m_CustomResolutionBox = dynamic_cast<GUICollectionBox *>(m_GUIControlManager->GetControl("CollectionCustomResolution"));
+		m_CustomResolutionBox->SetVisible(false);
+
+		m_CustomResolutionWidthTextBox = dynamic_cast<GUITextBox *>(m_GUIControlManager->GetControl("TextboxCustomWidth"));
+		m_CustomResolutionWidthTextBox->SetNumericOnly(true);
+		m_CustomResolutionWidthTextBox->SetMaxNumericValue(g_FrameMan.GetMaxResX());
+		m_CustomResolutionWidthTextBox->SetMaxTextLength(4);
+		m_CustomResolutionWidthTextBox->SetText(std::to_string(m_NewResX * g_FrameMan.GetResMultiplier()));
+
+		m_CustomResolutionHeightTextBox = dynamic_cast<GUITextBox *>(m_GUIControlManager->GetControl("TextboxCustomHeight"));
+		m_CustomResolutionHeightTextBox->SetNumericOnly(true);
+		m_CustomResolutionHeightTextBox->SetMaxNumericValue(g_FrameMan.GetMaxResY());
+		m_CustomResolutionHeightTextBox->SetMaxTextLength(4);
+		m_CustomResolutionHeightTextBox->SetText(std::to_string(m_NewResY * g_FrameMan.GetResMultiplier()));
+
+		m_CustomResolutionUpscaledCheckbox = dynamic_cast<GUICheckbox *>(m_GUIControlManager->GetControl("CheckboxCustomUpscaled"));
+		m_CustomResolutionUpscaledCheckbox->SetCheck(m_NewResUpscaled);
+
+		m_CustomResolutionBorderlessRadioButton = dynamic_cast<GUIRadioButton *>(m_GUIControlManager->GetControl("RadioBorderlessDriver"));
+		m_CustomResolutionDedicatedRadioButton = dynamic_cast<GUIRadioButton *>(m_GUIControlManager->GetControl("RadioDedicatedDriver"));
+		m_CustomResolutionApplyButton = dynamic_cast<GUIButton *>(m_GUIControlManager->GetControl("ButtonApplyCustomResolution"));
+		m_CustomResolutionMessageLabel = dynamic_cast<GUILabel *>(m_GUIControlManager->GetControl("LabelCustomResolutionValidation"));
+		m_CustomResolutionMessageLabel->SetVisible(false);
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void SettingsVideoGUI::SetEnabled(bool enable) const {
+		m_VideoSettingsBox->SetVisible(enable);
+		m_VideoSettingsBox->SetEnabled(enable);
+
+		if (m_CustomResolutionWidthTextBox->GetText().empty()) { m_CustomResolutionWidthTextBox->SetText(std::to_string(g_FrameMan.GetResX())); }
```

Comment:
Is it fine that this happens when enable is false? Before it used to only happen if enable was true.